src/providers/irc/AbstractIrcServer.cpp

```diff
@@ -23,6 +27,20 @@ AbstractIrcServer::AbstractIrcServer()
     this->writeConnection_->moveToThread(
         QCoreApplication::instance()->thread());
 
+    // Apply a leaky bucket rate limiting to JOIN messages
+    auto actuallyJoin = [&](QString message) {
```

Comment:
nit: this variable being named `message` feels weird.

Reply:
How else would you name it then?

Reply:
I'd suggest `channel` since that's what the message is.

Reply:
Originally, when I came up with the idea about this rate-limit system in my sleep I thought of it as something that could be used in multiple scopes, maybe outside of just `AbstractIrcServer`.

Reply:
On another note though, I think we shouldn't worry about something that might not end up being implemented/utilized at all. So maybe we can just change the name to `channel` as you're suggesting and maybe change to something more generic later if my planned idea ever sees the daylight.

Reply:
you send a message over a channel it doesn't really make sense to call a message that's being sent a channel, I think this is fine